apps/site/assets/ts/schedule/components/direction/GreenLineMenu.tsx

```diff
@@ -43,6 +72,19 @@ interface GreenLineItem {
   directionId: DirectionId;
 }
 
+const [state, dispatch] = useReducer(reducer, {
+  data: null,
+  isLoading: false,
+  error: false
+});
+
+const allRouteIds = ["Green-B", "Green-C", "Green-D", "Green-E"];
+let directionDestinations = new Map();
+for (var routeId in allRouteIds){
+  directionDestinations.set(routeId,fetchData(routeId, dispatch));
```

Comment:
I think it might only work if run inside a `useEffect()` callback. 😕 There's a similar example in `ScheduleModalContent`.

(I'm not sure how the for loop will play out! But the usual pattern is that the `fetchData` function will fetch the data and call the `dispatch` function, which will update the `state.data` with the fetched value. So in a loop you might end up overwriting the value with each iteration.)